src/main/scala/org/broadinstitute/dsde/firecloud/filematch/strategy/OntSingleReadStrategy.scala

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.dsde.firecloud.filematch.strategy
+
+import org.broadinstitute.dsde.firecloud.filematch.result.{FailedMatchResult, FileMatchResult, SuccessfulMatchResult}
+import OntSingleReadStrategy.FILE_CONTAIN_ENDINGS
+
+import java.nio.file.{Path, Paths}
+import scala.util.matching.Regex
+
+class OntSingleReadStrategy extends FileRecognitionStrategy {
+
+  override def matchFirstFile(path: Path): FileMatchResult = {
+    val fileName = path.getFileName.toString
+
+    // Used regex to match the file name containing the key and ending with any of the specified values
+    val foundMatch = FILE_CONTAIN_ENDINGS.find { case (key, values) =>
+      val regex = new Regex(s".*${Regex.quote(key)}.*(${values.map(Regex.quote).mkString("|")})$$")
+      regex.findFirstIn(fileName).isDefined
+    }
+
+    foundMatch match {
+      // we found a "read1"
+      case Some((_, values)) =>
+        // find the matching value
+        val value = values.find(value => fileName.endsWith(value)).get
+        // generate the id: strip the value from the filename.
+        val id = fileName.replace(value, "")
+        SuccessfulMatchResult(path, Paths.get(""), id)
+
+      // the file is not recognized
+      case None => FailedMatchResult(path)
+    }
+  }
+}
+
+object OntSingleReadStrategy {
+  // if the first file contains ${key} and ends with any of the values in the list, then this is a single read file
+  val FILE_CONTAIN_ENDINGS: Map[String, List[String]] = Map(
+    "Complete_barcode" -> List(".clean.fastq"),
+    "barcode" -> List(".fastq", ".fastq.gz")
+  )
+}
```

Comment:
I still don't think this is right. You may want to check with Adam to verify requirements - but the requirement as written in Jira is:

> … naming convention of some kind of \*barcode#####.*\.fastq or \*barcode#####.\*.fastq.gz pattern.

which means the only logic we should care about is:
* ends with `.fastq` or `.fastq.gz`
* contains `barcode` before the file extension (should this be case-sensitive or not?)

And we don't need any considerations for "Complete_" or ".clean". As written, this will not return the right results for input of "Repeat_barcode4.clean.fastq.gz", for instance.

Reply:
I have updated the matcher: https://github.com/broadinstitute/firecloud-orchestration/pull/1581/commits/315ac74b654e0eda3016592087cced3dd56dd72b and added more test cases to validate output: https://github.com/broadinstitute/firecloud-orchestration/pull/1581/commits/14012df22439dc3abcde2cb0bbf54b73d47300a7